Issue 8/old frontend refactor

lanvalird opened the pull request:
## Референсы

#8, #10, #11

## В кратце, что сделал

Покушал, попил водички, кофе... Ладно, немного переписал код, поменял где-то нейминг и фиксил параллельно баги, возникающие при ренейминге и переписи логики. Не все часы страдания были закомичены, но их итог да)))

## Ещё короче о главном

В большей степени переименовывал и перемещал файлы, выносил их логику в дочерние.

lanvalird committed:
-issue-10; -ref ranaming files to kebab-case

lanvalird committed:
-ref-up renaming icon files (`name-icon.tsx` -> `name.tsx`)

lanvalird committed:
-up add `icons/brands` folder

lanvalird committed:
-ref-add barriel exports for icons (re-export)

lanvalird committed:
-fix export and import icons

lanvalird committed:
-add `app` layer from fsd

lanvalird committed:
-add `shared` layer from fsd; -issue #11

lanvalird committed:
-fix; -issue #11

lanvalird committed:
-fix 2

lanvalird committed:
-ref-up renaming files

lanvalird committed:
-up-fix file's paths

lanvalird committed:
-up site header (props)

lanvalird committed:
-up site header (read in description)

Вынес часть логики по файлам, типы и вынес иконки в шапке в отдельные компоненты

lanvalird committed:
-add `entities` layer from fsd; -issue #8

lanvalird committed:
-add import aliases (#8 #11)

lanvalird committed:
-up logotype

lanvalird committed:
-ref files

lanvalird committed:
-fix

lanvalird committed:
-fix (надеюсь)

lanvalird committed:
-add member; -up Discord icon SVG

Так как в прошлый раз забыл поменять ветку, решил и этого добавить, но желаетльно слить, конечно

lanvalird committed:
-up formatLinks (from utils to another lib)

lanvalird committed:
-fix build

lanvalird committed:
-ref miniref

lanvalird committed:
-up dependencies

lanvalird commented on apps/frontend/src/app/app.tsx:
```diff
@@ -1,19 +1,20 @@
-import Header from './components/Header';
-import Footer from './components/Footer';
-import Loading from './pages/Loading';
+import { Header } from '@/shared/components/header';
+import { Footer } from '@/shared/components/footer';
+import Loading from '@/pages/loading';
+import { HEADER_LINKS } from './constants';
```
Надо отделить:
```ts
// Компоненты
import { Header } from '@/shared/components/header';
import { Footer } from '@/shared/components/footer';

// Страница (лучше создать файл loading.tsx в компонентах)
import Loading from '@/pages/loading';

// Файлы конфигурации
import { HEADER_LINKS } from './constants';
```

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/icons.tsx:
Возможно, стоит затронуть общий набор иконок, а не создавать отдельный файл только для theme-switching'а

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  );
+
+  useEffect(() => {
+    const mm = window.matchMedia('(prefers-color-scheme: dark)');
```
[понятное дело mm - это matchMedia, но...](https://docs.laf-team.ru/agreements/general#naming-6)

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,48 @@
+import { SystemThemeIcon, LightThemeIcon, DarkThemeIcon } from './icons';
+
+export const ThemeSwitcher = (props: React.ButtonHTMLAttributes<HTMLButtonElement>) => {
+  const [matches, setMatches] = useState(window.matchMedia('(prefers-color-scheme: dark)').matches);
```
`matches` чего? Темы? `themeMatches` тогда лучше

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  }, [matches, currentTheme]);
+
+  const toggleTheme = () => {
+    const themes = ['system', 'emerald', 'forest'];
```
лучше вынеси все эти темы и обращайся к какому-нибудь свойству объекта по ключу, как пример:
```ts
const themes = {
  system: 'system',

  dark: 'forest',
  light: 'emerald',

  // ...others
};

themes['light']; // 'emerald'
```

lanvalird commented on apps/frontend/src/shared/components/header/index.tsx:
```diff
@@ -0,0 +1,83 @@
+ */
+export const Header = (props: HeaderProps) => {
+  const [scrollY, setScrollY] = useState(0);
+  const ww = useDeviceWidth();
```
> Желательно избегать сокращений, исключение составляет общепринятая нотация, например: i, j, k в циклах, но даже их лучше писать полностью при возможности.

[Ссылка](https://docs.laf-team.ru/agreements/general#naming-6)

lanvalird commented on apps/frontend/src/shared/components/header/index.tsx:
```diff
@@ -0,0 +1,83 @@
+import { GitHubIcon, TelegramIcon } from '@/icons';
+import { ThemeSwitcher } from './theme-switcher';
+
+import clsx from 'clsx';
+import { Navigation } from './navigation';
+import { MobileHeader } from './mobile';
```
Забыл отсортировать :<

lanvalird commented on apps/frontend/src/shared/components/header/navigation.tsx:
```diff
@@ -0,0 +1,54 @@
+type Props = { links: HeaderNavLink[]; onClick?: () => void };
+
+export const Navigation = ({ links, onClick }: Props) => {
+  const ww = useDeviceWidth();
```
https://github.com/Lazy-And-Focused/site/pull/13/files#r2387927830

lanvalird commented on apps/frontend/src/shared/components/header/navigation.tsx:
```diff
@@ -0,0 +1,54 @@
+    return links.map((r) => (
+      <NavLink
+        key={r.path}
+        to={r.path}
```
r?

https://github.com/Lazy-And-Focused/site/pull/13/files#r2387927830

lanvalird commented on apps/frontend/src/shared/components/member-card/variants/default.tsx:
```diff
@@ -0,0 +1,43 @@
+import type { Member } from '@/entities/member';
```
забыл пропуск строки до типа

lanvalird commented on apps/frontend/src/shared/components/member-card/variants/default.tsx:
```diff
@@ -0,0 +1,43 @@
+              .filter((link) =>
+                Object.values(hasPrimarySocials).some((s) => link.href.startsWith(s[1])),
+              )
+              .map((s) => (
```
[s]ocial?

lanvalird replied:
Слишком много кода внутри `{}`

lanvalird commented on apps/frontend/src/shared/components/member-card/variants/default.tsx:
```diff
@@ -0,0 +1,43 @@
+        <h3 className='align-center flex flex-row items-center justify-end gap-x-2 text-base/7 font-semibold tracking-tight text-base-content'>
+          {member.name}
+
+          {hasPrimarySocials &&
```
[GitHub, блин](https://github.com/Lazy-And-Focused/site/pull/13/files#r2387948907)

FOCKUSTY commented on apps/frontend/public/images/logo.png:
Я так понимаю, ты взял одну из иконок. Однако это же старая версия, вроде даже градиент старый (если не ошибаюсь)

Вот новые: (также есть в Telegram)

<img width="512" height="512" alt="Оконный---светлая" src="https://github.com/user-attachments/assets/fb9f2b1d-4704-4129-bb26-cc931aec77a1" />

<img width="512" height="512" alt="Оконный--стандартная" src="https://github.com/user-attachments/assets/be115345-839a-46eb-a6c2-7568dd9bd367" />

lanvalird replied:
Это была второстепенная задача и я забыл, потому что ты за неделю уже третий ребрендинг делал

lanvalird reviewed (commented):
Ужас

FOCKUSTY reviewed (commented):
Что могу сказать... много ошибок и не соблюдения нашего соглашения. Жду правок от создателя (я ещё не всё заревьвил, где-то было лень)

lanvalird commented on apps/frontend/src/app/app.tsx:
```diff
@@ -1,19 +1,20 @@
+import Loading from '@/pages/loading';
+import { HEADER_LINKS } from './constants';
 
 /**
  * Корневой компонент приложения
  */
```
да

lanvalird commented on apps/frontend/src/app/index.tsx:
```diff
@@ -1,9 +1,10 @@
+
+import './styles/index.css';
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
```
Это стандартная практика с React-приложениями. Выносить root в отдельную константу нет смысла, так как приложение как ни крути привязано к index.html, который никто менять не будет))) В этом был бы смысл, если использовался шаблон с несколькими приложениями:

```html
<div id="app1"></div>
<div id="app2"></div>
```

Также, если элемент не будет найден, то ничего не произойдёт.

lanvalird replied:
*но я наверное всё же вынесу

FOCKUSTY replied:
Я по этому и сказал, что это не нужно

lanvalird commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -1,21 +1,26 @@
+  children?: ConfigRoute[];
+};
 
 /**
  * Все пути к страницам приложения (сайта)
  */
```
да.

lanvalird commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -1,21 +1,26 @@
-const Projects = lazy(() => import('../pages/Projects'));
-const NotFound = lazy(() => import('../pages/NotFound'));
-const Links = lazy(() => import('../pages/Links'));
+export type ConfigRoute = {
+  name?: string;
+  path: string;
+  element: React.ReactNode;
+  children?: ConfigRoute[];
+};
```
Именно, что усложнение кода, но в этом есть смысл, если бы были частые операции, где нужно было вытаскивать отдельный роут (таких нет)

lanvalird commented on apps/frontend/src/shared/lib/link-master/formater.ts:
```diff
@@ -0,0 +1,56 @@
+  };
+}
+
+/**
+ * Функция, преобразующая строковые ссылки в массив SocialLink[].
+ */
```
Я просто не придумал более строгое/подхлдящее название 😅

lanvalird commented on apps/frontend/src/pages/home.tsx:
```diff
@@ -72,12 +73,12 @@ const Home = () => {
+              .filter((member) => !member.meta?.includes('leave'))
+              .map((member) => (
+                <li key={member.tag}>
+                  <div>
+                    <TeamMemberCard member={member} />
                   </div>
```
Хз, не помню

lanvalird commented on apps/frontend/src/pages/home.tsx:
```diff
@@ -72,12 +73,12 @@ const Home = () => {
-                  <div key={person.tag}>
-                    <TeamMemberCard member={person} />
+            {members
+              .filter((member) => !member.meta?.includes('leave'))
```
> Скажу то же самое про `meta`: почему бы не сделать обязательным?

А зачем делать обязательным? Мы избавимся от "?" и всё? Мета не обязательна, потому что её, как правило, нет у обычного участника и служит для специальных возможностей...

lanvalird replied:
> Скажу то же самое про `meta`: почему бы не сделать обязательным?

А зачем делать обязательным? Мы избавимся от "?" и всё? Мета не обязательна, потому что её, как правило, нет у обычного участника и служит для специальных возможностей...

lanvalird commented on apps/frontend/src/pages/not-found.tsx:
```diff
@@ -2,8 +2,7 @@
 import { MouseParallaxChild, MouseParallaxContainer } from 'react-parallax-mouse';
 // -----
```
Название само за себя говорит

lanvalird commented on apps/frontend/src/shared/components/browser/index.tsx:
```diff
@@ -12,16 +18,8 @@ export type Tab = { id: string | number; name: string; favicon?: string; element
-}) => {
-  const [openTab, setOpenTab] = useState<Tab | null>(null);
+export const Browser = ({ title, tabs, children }: Props) => {
+  const [openTab, setOpenTab] = useState<BrowserTab | null>(null);
```
Я тогда был пьян 😅

lanvalird commented on apps/frontend/src/shared/components/browser/index.tsx:
> В одном месте ты использовал
{openTab?.element ?? children ?? 'Вкладка не выбрана'}, а в другом уже
{openTab ? openTab.name : 'Выберите вкладку'}, может привесит к единому стилю?

Так это разная логика. В первом случае мы используем элемент Таба, либо чилдрена, либо заполнитель. Во втором мы проверяем существует ли открытый таб (если под "один стиль" писать, то "сложный" для тебя код станет ещё сложнее за счёт "лишних" символов). Не знаю почему тебе таб кажется сложный, но это достаточно простой компонент, для менч больше вопросов возникает к компонентам MemberCard'а

lanvalird replied:
> В строке 52, ты сокращаешь tab до t, я думаю, что это плохо, ибо передаваемая функция в map не самая маленькая (tabs.map((t) => )

Она маленькая, просто раздута из-за классов... (Я переименовал)

lanvalird commented on apps/frontend/src/shared/components/browser/types.tsx:
```diff
@@ -0,0 +1,6 @@
+export type BrowserTab = {
+  id: string | number;
```
Обоснуй, что ты имеешь в виду под "единым типом'

lanvalird replied:
Зависит от ситуации. Сейчас я без проблем могу использовать число в качестве id, что полезно при работе с массивами и тд. Под капотом это всегда строка. Другими словами, пока технически это ни на что не влияет, то не лучше сделать, как будет удобно разработчику?

lanvalird committed:
-fix

lanvalird commented on apps/frontend/src/shared/components/footer/index.tsx:
```diff
@@ -26,5 +26,3 @@ const Footer = () => {
   );
 };
-
-export default Footer;
```
Он только отвлекает...

lanvalird committed:
-hot #13

lanvalird commented on apps/frontend/src/shared/components/header/mobile.tsx:
```diff
@@ -0,0 +1,56 @@
+  const modalRef = useRef<HTMLDialogElement>(null);
+
+  const showModal = () => {
+    if (modalRef && modalRef.current) {
```
Неправильный ответ! `null` никогда не будет равен `true`!!!

lanvalird commented on apps/frontend/src/shared/components/header/mobile.tsx:
```diff
@@ -0,0 +1,56 @@
+export const MobileHeader = ({ links }: HeaderProps) => {
+  const modalRef = useRef<HTMLDialogElement>(null);
+
+  const showModal = () => {
+    if (modalRef && modalRef.current) {
+      modalRef.current.showModal();
+    }
+
+    return null;
+  };
+
+  const closeModal = () => {
+    if (modalRef && modalRef.current) {
+      modalRef.current.close();
+    }
+
+    return null;
+  };
```
Это старый код, который было лень оптимизировать 🥱

(Логика Header ещё не была переписан, кроме как выноса _огромных_ компонентов)

lanvalird commented on apps/frontend/src/shared/components/header/navigation.tsx:
```diff
@@ -0,0 +1,54 @@
+      <NavLink
+        key={r.path}
+        to={r.path}
+        className={({ isActive, isPending }) =>
+          clsx(
+            'btn btn-ghost min-h-8 content-center text-nowrap rounded p-2 text-base font-medium text-base-content transition-colors sm:text-sm',
+            isActive && 'bg-primary/50',
+            isPending && 'animate-pulse',
+          )
+        }
```
Нет, это же целых 18 новых символов писать 😭

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,48 @@
+export const ThemeSwitcher = (props: React.ButtonHTMLAttributes<HTMLButtonElement>) => {
+  const [matches, setMatches] = useState(window.matchMedia('(prefers-color-scheme: dark)').matches);
+  const [currentTheme, setCurrentTheme] = useState(
+    localStorage.theme ?? (matches ? 'forest' : 'emerald'),
```
Не вижу смысла их выносить, но я бы вынес целиком логику определения темы

lanvalird commented on apps/frontend/src/shared/components/header/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,48 @@
+    setCurrentTheme(themes[(themes.indexOf(currentTheme) + 1) % themes.length]);
+  };
+
+  const icon =
+    currentTheme === 'system' ? (
+      <SystemThemeIcon />
+    ) : currentTheme === 'emerald' ? (
+      <LightThemeIcon />
+    ) : (
+      <DarkThemeIcon />
+    );
```
Твой код менее читабелен...

lanvalird commented on apps/frontend/src/shared/components/member-card/avatar/index.tsx:
```diff
@@ -16,12 +19,12 @@ type CardAvatarProps = {
   ...props
-}: CardAvatarProps & DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>) => {
+}: Props & DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>) => {
   const [isLoading, setLoading] = useState(true);
   const [isError, setError] = useState(false);
   const [imageLoaded, setImageLoaded] = useState(false);
```
Это невозможности. Оправдано.

lanvalird commented on apps/frontend/src/shared/components/member-card/contexts/index.tsx:
```diff
@@ -0,0 +1,7 @@
+import { createContext } from 'react';
+
+export type HasPrimarySocialsType = {
+  [key: string]: { 0: boolean; 1: string };
```
> Замени `{ 0: boolean; 1: string }` на кортеж `[boolean, string]`

Это не сработало...

lanvalird commented on apps/frontend/src/shared/components/member-card/index.tsx:
```diff
@@ -0,0 +1,85 @@
+}) => {
+  const validateSocial = useCallback(
+    (href: string) => {
+      const isExec = member.socials.some((link) => link.href.startsWith(href));
+      return isExec || false;
```
Хватит комментировать код, написанный на скорость 😡

lanvalird commented on apps/frontend/src/shared/components/member-card/index.tsx:
```diff
@@ -0,0 +1,85 @@
+    [validateSocial],
+  );
+
+  const [avatarSrc, setAvatarSrc] = useState(getAvatarUrl(member, hasPrimarySocials!['github'][0]));
```
А не ты ли говорил про modalRef и его true?

lanvalird committed:
-ref about.tsx

lanvalird committed:
-types

lanvalird committed:
-up deep components in about and home pages

lanvalird committed:
-up без комментариев (буквально)

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
+  /** Используется в шапке сайта */
   name?: string;
+
+  /** URI, как например `/home` */
```
URI, ну... нет. Больше location подходит

FOCKUSTY replied:
В таком случае, кстати, `path` можно назвать `location` и комментарий будет 100% бесполезный

FOCKUSTY commented on apps/frontend/src/pages/links.tsx:
```diff
@@ -5,6 +5,9 @@ import { TeamMemberCard } from '@/shared/components/member-card/index.tsx';
 import { teamSocialLinks } from '@/shared/config/team-social-links.ts';
 import { members } from '@/shared/config/lists/members.tsx';
 
+// Вынести, как и на/в других страницах
+const filteredMembers = members.filter((member) => !member.meta?.includes('leave'));
+
```
Ну так вынеси

FOCKUSTY commented on apps/frontend/src/shared/components/browser/index.tsx:
```diff
@@ -12,13 +12,16 @@ type Props = {
  */
-export const Browser = ({ title, tabs, children }: Props) => {
+export const Browser = ({
+  /** Заголовок окна */
+  title,
```
Зачем здесь комментарий? `title` буквально так и переводится
<img width="656" height="253" alt="image" src="https://github.com/user-attachments/assets/1d9dd886-53aa-44e7-9185-877554dc5fa2" />

FOCKUSTY commented on apps/frontend/src/shared/components/browser/index.tsx:
```diff
@@ -12,13 +12,16 @@ type Props = {
+export const Browser = ({
+  /** Заголовок окна */
+  title,
+  /** Вкладки с типом BrowserTab[] */
+  tabs,
```
`tabs` тоже переводится как "вкладки", а насчёт типа: его можно узнать, посмотрев на его тип, потому что у него тип `BrowserTab[]`

Меньше комментариев, особенно таких, они бесполезные

FOCKUSTY commented on apps/frontend/src/shared/components/browser/index.tsx:
```diff
@@ -29,7 +32,6 @@ export const Browser = ({ title, tabs, children }: Props) => {
 
       <div className='flex-1 overflow-y-auto p-4 text-gray-900 dark:text-gray-200'>
-        {/* Этот '&&' специальнр для @FOCKUSTY */}
         {(currentTab && currentTab.element) ?? children ?? 'Вкладка не выбрана'}
```
Эх...

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
 const NotFound = lazy(() => import('@/pages/not-found'));
 const Links = lazy(() => import('@/pages/links'));
 
+/**
+ * Конфигурация отдельного путя до страницы приложения.
+ */
```
Тоже лишние комментарии, они буквально так и называются, можешь ещё назвать `AppRouteConfig`, если тебе так нужна ясность

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
+ */
 export type ConfigRoute = {
+  /** Используется в шапке сайта */
   name?: string;
```
Лишний комментарий

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
   path: string;
+
+  /** Страница приложения (React-компонент)*/
   element: React.ReactNode;
```
А мы бы без комментария не догадались, что это React-компонент, у него же тип `any`

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
   element: React.ReactNode;
+
+  /** Вложенные ConfigRoute */
   children?: ConfigRoute[];
```
Да? Серьёзно?

Убери комментарии

lanvalird replied:
Нет.

lanvalird replied:
> Да? Серьёзно?
> 
> Убери комментарии

Возьму сам себя на слово: в следующем ревью буду попросить объяснить КАЖДОЕ действие, которое ты сделал, если не пойму (а я наверное не пойму) и почему ты поступил именно так... Хотя, так и должно произваодиться ревью, но может ты поймёшь мою боль, когда я буду докапываться до каждой мелочи, которая "ни на что и не влияет"

FOCKUSTY commented on apps/frontend/src/app/routes/index.tsx:
```diff
@@ -10,15 +10,27 @@ const Projects = lazy(() => import('@/pages/projects'));
+ *
+ * @returns {ConfigRoute[]} ConfigRoute[]
  */
 export const routes: ConfigRoute[] = [
```
Правда? Я бы не догадался, что это массив

Убирай комментарии

FOCKUSTY commented on apps/frontend/src/shared/components/browser/index.tsx:
```diff
@@ -12,13 +12,16 @@ type Props = {
- *
  * @example <Browser title="Браузер" tabs={tabs} />
  * @example <Browser title="Браузер" tabs={tabs}>Заполнитель, если не выбрана вкладка</Browser>
  */
```
До сих пор не вижу смысла в этой js-доке, можно убрать и ничего не измениться

FOCKUSTY reviewed (commented):
За эти 4 коммита я понял, что Валя понял, что такое комментарии и научился их писать, но их писать не нужно
В этом коде комментарии не нужны, он понимается и без них. Даже я в своём их не пишу, хотя мой код менее понятный. Уберешь комментарии и ок
Вроде это не все изменения, которые должны быть, но конкретно эти прошли проверку, но апрувать боюсь

FOCKUSTY commented:
Некоторые мои ревью поехали...

lanvalird commented:
У меня GitHub лагает, поэтому могу ответить только таким образом:

---

> Что за бред?

Костыли, и кажется, вынужденная мера. Что ты хочешь от участка кода, который ещё не переписывался? Реквест на ревью я не отправлял 

---

> У тебя функция, а не метод класса, так что лучше называть четко без привязки к "абстрактному классу"
> 
> Изменено 4:58 по МСК 01.10.2025
> Вроде я говорил про это:
> ```ts
> export const formatMany = (links: string[]): SocialLink[] => {
>   return links
>     .map((link) => format(link))
>     .sort((a, b) => (a.special === b.special ? 0 : a.special ? -1 : 1));
> };
> ```

Эта конкретика лишняя. Функция расположена в специальном каталоге линк-мастер (точнее URL ссылок), и если надо, то можно переименовать отдельно в файлах. Привожу пример с пакетами MDX: есть метод compile... Тебе ясно, что он делает? Здесь та же логика. Ещё как пример: в некоторых библиотеках/фреймворках есть хуки, подобные useFetcher или useAuth. Разве неясно, что они делают? Если со вторым понятно, что скорее всего используется глобальное состояние приложения для того, чтобы взять ауф,  то юзФетчер надо смотреть из контекста. Объявленный в компоненте Project он будет фетчить проект, объявленный в каталоге `users/` он будет фетчить пользователя или пользователей.

lanvalird commented:
Я согласен про места, где в комментариях были описаны типы (привычка с разработки API под JS), но про удаления их я против, если честно.

lanvalird commented:
По поводу этого участка:
```
export const formatMany = (links: string[]): SocialLink[] => {
  return links
    .map((link) => format(link))
    .sort((a, b) => (a.special === b.special ? 0 : a.special ? -1 : 1));
};
```

Валя, у тебя функция "не чистая". Она должна форматировать, а не сортировать

lanvalird commented on apps/frontend/src/shared/lib/link-master/formater.ts:
```diff
@@ -0,0 +1,56 @@
+ * @param input Строка для очистки
+ * @returns {string} string
+ */
+function getClearedLink(input: string): string {
```
А что не так? Просто функция "очищающая" ссылку. Ссылку можно по разному почитсить: убрать протокол, убрать хостнейм оставив только URI/URN, и тд...

lanvalird replied:
> Ссылку можно по разному почитсить: убрать протокол, убрать хостнейм оставив только URI/URN, и тд...

Поэтому я против того, чтобы убирать комментарии. Это только на первый взгляд кажется, что нормально. Кстати, мои комментарии сейчас низкого качесвта, но это не говорит о том, что они бесполезные

lanvalird commented on apps/frontend/vite.config.ts:
Не помню, что там было, но вроде я просто настроил под новую архитектуру файлик. В частности, модули, которые будут заранее "запечены"

lanvalird commented:
https://github.com/Lazy-And-Focused/site/pull/13#discussion_r2391307161

Скорее всего, где loaded более новый код, а до старого ещё просто не дошёл. Моя главная задача в исью была перенести файлы по папкам 🤷‍♂️

К остальному это уже отдельная тема

lanvalird commented:
> Я комментирую всё, я не знаю, что было написано на скорость, а что нет
>
> ПЛОХО писать код на скорость, у тебя нет дедлайнов

ничё не знаю, тогда были

lanvalird committed:
-fix-up useDeviceWidth to @shared/hooks/index.ts; app script in to outside file

lanvalird committed:
-up link-master (+rename)

lanvalird committed:
-up member-card

lanvalird committed:
-up app-header/navigation

lanvalird committed:
-up tsconfig paths; -ref src/app

lanvalird committed:
-ref src/shared/config

lanvalird committed:
-ref member-card; -fix about and links pages

lanvalird committed:
-up comments and sorting imports

lanvalird committed:
-up app router module

lanvalird committed:
-up members info

lanvalird committed:
-ref format

lanvalird committed:
-fix-up

lanvalird committed:
-fix-up (oops :M)

lanvalird committed:
-up

lanvalird committed:
-add APP_ROUTES constant in @app/constants.ts <_<

lanvalird committed:
-up home page

lanvalird committed:
-ref homepage; -move file constants.ts

lanvalird committed:
-ref about page

lanvalird committed:
-up loading page

lanvalird committed:
-ref links page

lanvalird committed:
-up-aboba

lanvalird committed:
-format

lanvalird committed:
-up imports at `about/sections/credits.tsx`

lanvalird committed:
-ref not-found page

Commit writed in 90% with AI

lanvalird committed:
-ref projects page; -ref browser components

lanvalird committed:
-fix yes

lanvalird committed:
-format

lanvalird committed:
-ref app-footer (mini-mini-miniref, nya!)

lanvalird committed:
-meow (-move loading component)

lanvalird committed:
-fix

lanvalird committed:
-add shared hook useFilteredMembers

lanvalird committed:
-format

lanvalird committed:
-up project page repos constant

lanvalird committed:
-fix

lanvalird committed:
-add errorCatched in use-projects hook into projects page

lanvalird committed:
-fix constants

lanvalird committed:
-ref app-header's miniref

lanvalird commented on apps/frontend/src/shared/components/app-header/index.tsx:
_Не забыть вынести компоненты_

lanvalird commented on apps/frontend/src/shared/components/member-card/variants/full.tsx:
_Разложить на подкомпоненты_

lanvalird commented on apps/frontend/src/shared/components/member-card/index.tsx:
_Разложить на дочерние компоненты_

lanvalird commented on apps/frontend/src/shared/lib/constants.ts:
```diff
@@ -0,0 +1,31 @@
+];
+
+// В будущем можно добавить (для фетчинга с бэка, например):
+// export const API_BASE_URL = import.meta.env.VITE_API_URL;
```
Если добавлять, то:
```ts
// если в ENV не окажется переменной
export const API_BASE_URL = import.meta.env.VITE_API_URL || import.meta.url; // или `${import.meta.url}/api`
```

lanvalird reviewed (commented):
Окей, надо подправить сортировку импортов (хуки должны быть первее компонентов) и переписать шапку сайта с компонентом участника — _для меня ключевое_

FOCKUSTY commented on apps/frontend/src/pages/home/lib/constants.ts:
```diff
@@ -0,0 +1,10 @@
+import { ORGANIZATION_SHORT_NAME } from '@shared/lib/constants';
```
Ну мы ещё не организация, но ошибки тут нет

lanvalird replied:
Я использовал обобщенный термин. Опенсорс, мало ли

FOCKUSTY commented on apps/frontend/src/pages/links/hooks/use-copy-link.ts:
```diff
@@ -0,0 +1,24 @@
+ * Хук для копирования ссылки на _текущую_ страницу
+ */
+export const useCopyLink = () => {
+  const copyButtonRef = useRef<HTMLButtonElement>(null);
```
В данном случае у нас имеется контекст, так что нельзя просто назвать `ref`?

lanvalird replied:
Просто раньше этот звук был в компоненте, а реф не был переименован... +Это опять что-то из категории про ObjectRoute... С одной стороны да, с другой неясно, а не нарушаем ли мы кодовое соглашение. Для меня вот контекст ясен, но крайне неудобно видеть функцию:
1. useCopy() — непонятно для чего (время придраться к документации!)
2. ref чего? Почему я сразу же не могу узнать для чего он, почему мне надо скроллить вниз, чтобы понять?
3. Но если учитывать всё выше написанное, включая твой ревью, то можно переименовать в actionButtonRef (buttonRef всё ещё чересчур обобщенно)

Возможно, притянул за уши.

FOCKUSTY commented on apps/frontend/src/pages/links/lib/constants.ts:
```diff
@@ -0,0 +1,15 @@
+      DESCRIPTION: 'Если вы тут впервые, но вам интересна команда, то глядите!',
+    },
+  },
+} as const;
```
Я насчёт `as const`, его также можно указать у значений нашего объекта, а то его так можно изменять, если не ошибаюсь

lanvalird replied:
Я специально проверял — нельзя. По-крайней мере Visual Studio подсвечивал ошибкой :)

*Компилятор не компилил TS, кстати, поэтому должно работать, но я проверю, если не забуду, ещё раз

FOCKUSTY replied:
Нельзя? Я дофига раз так делал, теперь интересно даже

FOCKUSTY commented on apps/frontend/src/pages/projects/index.tsx:
```diff
@@ -0,0 +1,26 @@
+      <ProjectsHero>
+        <Browser tabs={tabs}>
+          {!errorCatched && (!loaded ? loadingPlaceholder : placeholder)}
+          {errorCatched && 'Произошла ошибка при загрузке данных'}
```
Хотелось бы, чтобы это было вынесено, но не обязательно (хотя мне сильно не нравится)

lanvalird replied:
Пока что это наилучшее решение, которое я придумал

FOCKUSTY commented on apps/frontend/src/pages/projects/lib/constants.ts:
```diff
@@ -0,0 +1,11 @@
+  },
+  BROWSER: {
+    /** Browser's placeholder */
+    EMPTY_TEXT: 'Выбери любой проект из списка выше',
```
`EMPTY_TEXT` -> `PLACEHOLDER`?

FOCKUSTY reviewed (changes requested):
(я не знаю, какие коммиты я ревьювил)
Я забыл, что делал и к чему претензии
Могу сказать, что некоторые константы можно вынести

lanvalird commented on apps/frontend/src/app/providers/router/routers/app.tsx:
```diff
@@ -0,0 +1,19 @@
+    element: <NotFoundPage />,
+  },
+];
+export { routes as appRoutes };
```
Потому что это не константа, а именно что конфигурационный объект.  +Такая константа уже есть

lanvalird commented on apps/frontend/src/app/providers/router/routers/main.tsx:
```diff
@@ -0,0 +1,12 @@
+  { name: 'Проекты', path: APP_ROUTES.PROJECTS, element: <ProjectsPage /> },
+  { name: 'О нас', path: APP_ROUTES.ABOUT, element: <AboutPage /> },
+  { name: 'Ссылки', path: APP_ROUTES.SOCIALS, element: <LinksPage /> },
+];
```
Название как раз-таки соответствует типу. Ты внимательнее посмотри код.

lanvalird replied:
> Обнв: с самим неймингом папко проблем не вижу {...}
> 
> Другими словами, всё названо верно и если цепляться за нейминг, то надо учитывать контекст, который у тебя есть (а он есть, GitHub отображает пути + сам файл роутера у тя тоже есть)

Но можно создать внутри `routers` папку/каталог `subs` или `sub-routers`, переместив  внутрь App и Main, а в родительном создать файл `index.ts`

lanvalird replied:
> + это же константа? Почему не ROUTES?

Возникали проблемы с типизацией и чтобы не мучаться оставил как есть, если говорить про константу. С другой стороны, это в таком случае уже не константа... Всё сложно

> Я разрешаю не переименовывать и оставить как есть, теперь понял, спасибо <3

Резолвну это обсуждение, когда внесу правки с index.ts (а то его нет, что немного не оч)

lanvalird commented on apps/frontend/src/app/providers/router/utils.tsx:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Конвертирует AppRoute в RouteObject для react-router
+ */
+export const convertToRouteObjects = (routes: AppRoute[]): RouteObject[] => {
```
Ты попался на свою же ловушку. Там указан тип

lanvalird commented on apps/frontend/src/pages/home/sections/hero.tsx:
```diff
@@ -0,0 +1,33 @@
+  description?: string;
+  background?: string;
+  children?: React.ReactNode;
+}
```
Вроде можно

lanvalird commented on apps/frontend/src/pages/links/hooks/use-copy-link.ts:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Хук для копирования ссылки на _текущую_ страницу
+ */
+export const useCopyLink = () => {
```
Это предложение из категории "из чего конвертируем в ObjectRoute"? Вообще можно использовать просто useCopy, но тогда боюсь ревьювер уже придирётся к тому, что не поймёт, что мы копируем (а вдруг это зук копирующий массив, idk)

lanvalird commented on apps/frontend/src/pages/links/sections/team-members.tsx:
```diff
@@ -0,0 +1,35 @@
+
+export const TeamMembersSection = () => {
+  const filteredMembers = useFilteredMembers();
+  const { TEAM_MEMBERS: TEAM_CONSTANTS } = LINKS_PAGE_CONSTANTS.SECTIONS;
```
Догадайся... У нас есть в коде такая константа, содержащая участников команды. То есть во из избежании путаницы.

FOCKUSTY replied:
Я её не увидел, вот и задал вопрос

lanvalird commented on apps/frontend/src/pages/not-found/lib/constants.ts:
```diff
@@ -0,0 +1,46 @@
+      'Наш пушистик прошерстил весь сайт, но даже его шустрые лапки с острыми коготками не сумели закотить эту страницу!',
+    buttonText: 'Na glavnuyu',
+  },
+} as const;
```
Видимо лень

FOCKUSTY replied:
Development in LAF be like:

lanvalird commented on apps/frontend/src/pages/projects/hooks/use-projects.ts:
Компоненты я стараюсь объявляиь по старому, в утилитах стараюсь писать под неанонимные

lanvalird commented on apps/frontend/src/pages/projects/ui/browser/components/browser-dropdown-item.tsx:
```diff
@@ -0,0 +1,25 @@
+    <button
+      className={`relative flex w-full min-w-max items-center gap-4 overflow-hidden text-ellipsis text-nowrap px-4 py-2 text-left hover:bg-gray-100 dark:hover:bg-gray-800 ${
+        selected ? 'bg-gray-100 dark:bg-gray-800' : ''
+      }`}
```
Нет

lanvalird commented on apps/frontend/src/shared/hooks/use-filtered-members.ts:
```diff
@@ -0,0 +1,18 @@
+  meta.forEach((meta) => (members = members.filter((member) => !member.meta?.includes(meta))));
+
+  return members;
+};
```
Тут по факту, я поленился

lanvalird committed:
-fix

lanvalird committed:
-fix constants in not found page

lanvalird committed:
-ref projects page

FOCKUSTY commented on apps/frontend/src/pages/about/sections/credits.tsx:
```diff
@@ -0,0 +1,99 @@
+    <CreditsSection
+      id={'copyright'}
+      title={ABOUT_PAGE_CONSTANTS.SECTIONS.COPYRIGHT.TITLE}
+      description={ABOUT_PAGE_CONSTANTS.SECTIONS.COPYRIGHT.DESCRIPTION}
```
`ABOUT_PAGE_CONSTANTS.SECTIONS.COPYRIGHT.DESCRIPTION` слишком длинно =(

lanvalird committed:
-up (-ref?) app router

lanvalird committed:
-fix filter members function

lanvalird committed:
-fix

lanvalird committed:
-up header from #20 (premade)

lanvalird committed:
-oops

lanvalird committed:
-add new__app-header (temporarily)

lanvalird committed:
-add new__app-header

lanvalird committed:
-up styles in new header

lanvalird committed:
-add `AppHeaderLogotype` into new__app-header

lanvalird committed:
-up new__app-header

Add:

- AppHeaderNavigation
- `types.ts` into folder `new__app-header`

Move from file: AppHeaderLogotype

lanvalird committed:
-fix z-index for new__app-header

lanvalird committed:
add `AppHeaderNavigationItem`

lanvalird committed:
-up header navigation item

Resolve #22

lanvalird committed:
-fix build

lanvalird committed:
-up app-header navigation item styles

Resolve #22

lanvalird committed:
-add AppHeaderSocialLinks into new__app-header; -add HEADER_SOCIAL_LINKS

lanvalird committed:
-issue 23

Resolve #23

lanvalird committed:
-add new header hook into new__app-header; -fix styles

lanvalird committed:
-add modal app header without logic...

lanvalird committed:
-fix modal close in app header

lanvalird committed:
-fix open and closes app header modal

lanvalird committed:
-issue #25

Resolve #25
Resolve #26 (забыл закрыть до этого)

lanvalird committed:
-up new__app-header (add catalog 'variants')

lanvalird committed:
-ref rename new__app-header > app-header

lanvalird committed:
-format

lanvalird committed:
-review from @lanvalird

lanvalird committed:
-review from @FOCKUSTY

lanvalird committed:
-format

lanvalird committed:
-fix types into app header cmponent

lanvalird committed:
Merge pull request #27 from Lazy-And-Focused/issue-20/old-frontend-header-refactor

-issue #20

lanvalird committed:
-fix swap/switch mobile and default headers

lanvalird committed:
-add new icons

lanvalird committed:
-issue #19

Resolved #19

lanvalird committed:
-add new member-card

lanvalird committed:
-fix

lanvalird committed:
-fix avatars from members and into member-card

lanvalird committed:
-ref clean member-card

lanvalird committed:
-fix

lanvalird committed:
-add new member-card variant

lanvalird committed:
-format

lanvalird committed:
-up member-card component

lanvalird committed:
-up login in member-card component

lanvalird committed:
-up styles in member-card component

lanvalird committed:
-up styles in member-card component

lanvalird committed:
-up member-card (add socials modal)

lanvalird committed:
-up modals (merged from header & member-card)

lanvalird committed:
-fix

lanvalird committed:
-up `constants.ts` in frontend

lanvalird committed:
-up member-card: sub components moved

lanvalird committed:
-fix falsy socials count in member card

lanvalird committed:
Merge branch 'issue-8/old-frontend-refactor' into issue-21/old-frontend-member-card-refactor

lanvalird committed:
Merge pull request #29 from Lazy-And-Focused/issue-21/old-frontend-member-card-refactor

-issue 21

lanvalird committed:
-format-style use-shift-position

lanvalird committed:
-up export styles app-footer to `STYLE` const

lanvalird committed:
-up export styles for page into `STYLE` const

lanvalird committed:
-format

lanvalird committed:
-oops

lanvalird committed:
-ref-up export styles from app-footer

lanvalird committed:
-ref-up export many styles in header components

lanvalird committed:
-ref-up export styles from app-footer to files

lanvalird committed:
-ref-up export styles from member-card to files

lanvalird committed:
-format

lanvalird committed:
-up workspace's json to frontend

lanvalird committed:
-ref-up export styles from about page to files

lanvalird committed:
-ref-up export styles from home page to files

lanvalird committed:
-fix styles into constants (BASE -> CONTAINER)

lanvalird committed:
-ref-up export styles from loading page to files; -ref links page styles

lanvalird committed:
-format

lanvalird committed:
-ref-up export loading page styles to files

lanvalird committed:
-ref-up export links page styles to files (socials-links section)

lanvalird committed:
-fix non-exported styles

lanvalird committed:
-fix styles

lanvalird committed:
-ref-up export projects page styles to files from browser ui

lanvalird committed:
-up add hint-comment

lanvalird committed:
-update html metatags and manifest

lanvalird committed:
chore(frontend/pages(About)): fix reusable styles

lanvalird committed:
chore(frontend/pages(Home)): fix reusable styles

lanvalird committed:
chore(frontend): patch package version

lanvalird committed:
chore(frontend): format files

lanvalird committed:
refactor(frontend): resolve review from #33

lanvalird committed:
chore(frontend): format

lanvalird committed:
chore(frontend): rename LINK_ICON to LINK.ICON into MemberCardModal

lanvalird committed:
Merge pull request #33 from Lazy-And-Focused:issue-17/export-styles-to-files

refactor(frontend): вынос стилей в константы и исправление ошибок

lanvalird committed:
chore(frontend): rename `convertToRouteObjects` to `convertAppRoutesToRouteObjects` function in `router/utils` (app layer)

lanvalird committed:
refactor!(frontend): remove browser for projects page (legacy code)

lanvalird committed:
chore(frontend): format files

lanvalird committed:
chore(frontend/features): export local useCopy into links page to
`@features/copy-link`

lanvalird committed:
chore(frontend): fix unused import

lanvalird committed:
chore(frontend): delete untracked files

lanvalird committed:
chore(frontend): run `pnpx update-browserslist-db@latest`

lanvalird committed:
refactor!(frontend): migrate member parts to single `@entity/member`

BREAKING CHANGES:
- rename hooks: remove prefix `use-` in file names
- member ui remove to `@entity/member` folder

lanvalird committed:
chore(frontend): fix imports

lanvalird committed:
chore(frontend): update dependencies

lanvalird committed:
chore(frontend): migrate from `vite-tsconfig-paths` to native vite
features

lanvalird committed:
chore(frontend): from `@vitejs/plugin-react-swc` to
``@vitejs/plugin-react` (vite recomendation)

lanvalird committed:
fix(frontend/vercel): fix import into vite config file

lanvalird committed:
fix(frontend): compile TailwindCSS in dev mode Step 1 (from 2)

lanvalird committed:
feat(frontend): add theme switcher with light/dark mode toggle

lanvalird committed:
fix(frontend): resolve Tailwind CSS v4 reference and missing component files

lanvalird committed:
fix(frontend): compile TailwindCSS in dev mode Step 2 (from 2)

lanvalird committed:
chore(frontend): reconfig styles and fix tsx code

lanvalird committed:
fix(frontend): improve logotype container width and clean up theme hook implementation

- Fixed logotype container to have a maximum width to prevent layout overflow
- Simplified theme hook by removing redundant class manipulation and improving code clarity
- Maintained functionality while reducing unnecessary DOM operations

lanvalird committed:
chore(frontend): format files

lanvalird committed:
chore(frontend/deps): update dependencies and fix URL formatting

- Updated @tailwindcss/postcss from ^4.2.1 to ^4.2.2
- Updated tailwindcss from ^4.2.1 to ^4.2.2
- Updated vite-plugin-vercel from ^11.0.1 to ^11.0.3
- Fixed URL formatting in package.json by removing trailing slashes from homepage and author URL

lanvalird committed:
refactor(frontend): remove vite-plugin-vercel dependency and configuration

Removed the vite-plugin-vercel package and its configuration from the frontend application. This change simplifies the build setup by removing the Vercel-specific plugin that was previously used for deployment optimizations. The application will now use standard Vite configuration without the additional Vercel plugin layer.

lanvalird committed:
fix(frontend): replace react-router-dom with react-router

lanvalird committed:
chore(frontend): update primary content color in forest theme

Updated the primary content color in the forest theme from oklch(0.89 0.02 151.46) to oklch(0.92 0.02 151.55) for improved contrast and readability.

lanvalird committed:
chore(frontend): disable default theme for `forest` in daisyui configuration

Removed the default theme flag from the forest theme configuration in globals.css to prevent it from being automatically applied as the default theme.

lanvalird committed:
chore(frontend/docs): fix typo in JSDoc for filtered-members hook

lanvalird committed:
refactor(frontend/member-card): extract UI components and simplify layout

Extract MemberCardName, MemberCardDescription, and MemberCardSocials into dedicated UI components to improve modularity and separation of concerns. Remove the inline SocialLinksRow helper and simplify the main card rendering logic. Update barrel exports and adjust modal styles for consistency.

lanvalird committed:
chore:(frontend/deps) bump version to 1.4.1 and update frontend dependencies

- Upgrade react-router, eslint, typescript-eslint, and vite to latest versions
- Update pnpm-lock.yaml to reflect dependency changes
- Refactor member card modal button styling to use DaisyUI utility classes for improved visual consistency

lanvalird commented on apps/frontend/src/entities/member/card/ui/description/index.tsx:
```diff
@@ -0,0 +1,5 @@
+import { STYLE } from './style';
+
+export const MemberCardDescription = ({ description }: { description: string }) => {
+  return <p className={STYLE.DESCRIPTION}>{description}</p>;
```
`STYLE.MAIN` или подобно этому, думаю

lanvalird reviewed (commented):
Заревьювил пару файлов для правок. Также стоит использовать `RefObject` вместо/заместо `MutableRefObject`. Также можно заменить возможные `forwardRef` на обычные пропсы (это позволяет React 19)

lanvalird commented on apps/frontend/src/entities/member/hooks/filtered-members.ts:
Очень запутано и стоит подумать, как спроектировть кодовоую базу так, чтобы не оборочивать эту функцию в хук (это бессмысленно)

lanvalird commented on apps/frontend/src/entities/member/hooks/filtered-members.ts:
```diff
@@ -0,0 +1,30 @@
+  }
+  const members = [...grantedMembers];
+
+  return filterMembersByWithoutMeta(members, meta);
```
`members` можно передать напрямую

lanvalird commented on apps/frontend/src/entities/member/hooks/filtered-members.ts:
```diff
@@ -0,0 +1,30 @@
+   */
+  grantedMembers: Member[] = TEAM_MEMBERS,
+) => {
+  const meta = [...grantedMeta];
```
Можно использовать `Array.copy()`. Такой вариант написания очевиднее и не рябит в глазах (дело вкуса).

---

Предложение из чата от @FOCKUSTY

lanvalird commented on apps/frontend/src/entities/member/lib/utils.ts:
```diff
@@ -0,0 +1,17 @@
+) {
+  let result = [...grantedMembers];
+
+  grantedMeta.forEach((meta) => {
+    result = result.filter((member) => {
+      return !member.meta?.includes(meta);
+    });
+  });
```
## Разделение на дочерние функции

Не обязательно, но можно добавить "пояснительную" переменную, чтобы упростить восприятие кода:

```ts
grantedMeta.forEach((meta) => {
    result = result.filter((member: Member) => {
      const metaExcited = member.meta?.includes(meta);
      return !metaExcited;
    });
});
```

А лучше разделить:

```ts
grantedMeta.forEach((meta) => {
	function isMetaExcited(member: Member) {
    const metaExcited = member.meta?.includes(meta);
    return !metaExcited;
  }

  result = result.filter(isMetaExcited);
});
```

Но второй вариант может выглядеть излишнем в подобных строках.

## Использование более внятного имени для `meta`

Название переменной `meta` недостаточно конкретно и вызывает путаницу. Следует придумать новое.

lanvalird commented on apps/frontend/src/entities/member/entity.ts:
```diff
@@ -0,0 +1,20 @@
+// Есть идея добавить поля "skills". Также в бэкенде
+// есть свои типы,поэтому следует потом привести всё
+// к одному стандарту. Короче, открыть issue надо
+// будет и стереть этот комментарий.
```
Можно удалить, если поле `skills` ещё не добавили. Как ни крути, но комментарий вызывает вопросы: "Почему он всё ещё тут?"

lanvalird commented on apps/frontend/src/entities/member/entity.ts:
```diff
@@ -0,0 +1,20 @@
+  avatar?: string;
+  description?: string;
+  socials?: string[];
+  meta?: string[];
```
Есть смысл использовать тип `Array<string>`, а не `Array<string | undefined>`. Почему? Массивы легче проверять по количеству элементов, чем по существу (`array !== undefined`).

Это относится и к `socials`, и к `meta`.

lanvalird commented on apps/frontend/src/entities/member/entity.ts:
```diff
@@ -0,0 +1,20 @@
+// к одному стандарту. Короче, открыть issue надо
+// будет и стереть этот комментарий.
+
+export type BaseMember = {
```
BaseMember допустимое название, но я бы предпочёл что-то более конкретное: `RawMember` ("сырой")

lanvalird commented on apps/frontend/src/entities/member/index.ts:
Допустимо, но лучше не допускать "*", контролируя экспорты самому

lanvalird commented on apps/frontend/src/features/copy-link/index.ts:
Вероятно, не стоит менять текст в хуке, а вынести это. Это необязательно, но хочется, чтобы не хук контролировал состояние компонента, а компонент.

Если это необходимо, то можно создать `CopyButton` элемент и подобные при необходимости и вынести всю необходимую логику.

lanvalird commented on apps/frontend/src/pages/about/ui/credits-section/index.tsx:
В этом компоненте можно сократить количество переносов между HTML-тегами

lanvalird commented on apps/frontend/src/pages/projects/ui/projects-hero/style.ts:
```diff
@@ -0,0 +1,21 @@
+    'flex-col items-center justify-center',
+    'px-6 pt-14 lg:px-8',
+    'bg-cover bg-center bg-no-repeat',
+    // STATIC URL!??!?!?
+    // @TODO: Export static url to constant
+    "bg-[#418d76] bg-[url('/images/backgrounds/green.png')]",
```
Выполнить `@TODO`

lanvalird commented on apps/frontend/src/shared/components/app-header/hooks/use-shift-position.ts:
Возможно стоит подумать над таким вариантом:

```ts
useEffect(() => {
    const handleScroll = () => {
      return setShift(window.scrollY);
    };

    window.addEventListener('scroll', handleScroll);
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);
```

...чтобы отделить функцию от обработчиков (подписка/отписка)

lanvalird commented on apps/frontend/src/shared/components/app-header/ui/logotype/index.tsx:
```diff
@@ -0,0 +1,31 @@
+};
+
+export const AppHeaderLogotype = ({
+  size = 32,
```
Хорошо, что вынес всё остальное, но это тоже: `DEFAULT_IMAGE_WIDTH` или `DEFAULT_IMAGE_SIZE`.

lanvalird commented on apps/frontend/src/shared/components/base-modal/index.tsx:
Очень разрозренная вёрстка, если сравнивать с другими элементами. Можно вынести дочерние элементы, чтобы уравнять.

lanvalird commented on apps/frontend/src/shared/components/ui/icons/index.tsx:
Согласен с фокусти про лишние комментарии, так как `./brands` уже очевидно, что каталог, пока те два про конкретные иконки.

lanvalird commented on apps/frontend/src/shared/components/ui/theme-switcher/icons.tsx:
Лучше вынести в общий сет с иконками

lanvalird commented on apps/frontend/src/shared/components/ui/theme-switcher/index.tsx:
```diff
@@ -0,0 +1,13 @@
+  const { theme: _theme, toggleTheme, isDark } = useTheme();
+
+  return (
+    <button onClick={toggleTheme} className='btn btn-ghost btn-circle' aria-label='Toggle theme'>
```
Стили не вынесены.

lanvalird commented on apps/frontend/src/shared/hooks/theme.ts:
Названия для светлой и тёмной тем вынести в константы ОБЯЗАТЕЛЬНО.

lanvalird commented on apps/frontend/src/shared/components/app-header/ui/logotype/constants.ts:
```diff
@@ -0,0 +1,6 @@
+
+export const ORGANIZATION_NAME = APP_ORGANIZATION_NAME;
+
+export const DEFAULT_IMAGE_SOURCE = '/images/logo.png';
```
Вот где тут нужно добавить `DEFAULT_IMAGE_SIZE`.

lanvalird commented on apps/frontend/src/shared/components/app-header/ui/logotype/constants.ts:
```diff
@@ -0,0 +1,6 @@
+import { ORGANIZATION_NAME as APP_ORGANIZATION_NAME } from '@/shared/lib/constants';
+
+export const ORGANIZATION_NAME = APP_ORGANIZATION_NAME;
```
Это странно, но ладно.

lanvalird commented on apps/frontend/src/shared/components/app-header/variants/index.tsx:
Разнести варианты компонента по своим файлам.

lanvalird reviewed (commented):
Саморевью подошло к завершению. Дополнил предыдущие. В основном вопросы стилистические. Сильно в код не вникал. Путанно. Всё _(конец ревью)_.

lanvalird committed:
chore(frontend): simplest cleanup

lanvalird committed:
chore(deps): to up pnpm version

lanvalird committed:
chore(deps/frontend): update dependencies
